theano/gpuarray/tests/test_dnn.py

```diff
@@ -2298,3 +2298,233 @@ class Cudnn_grouped_conv(Grouped_conv_noOptim):
     conv2d_gradi_op = dnn.GpuDnnConvGradI
     flip_filter = False
     is_dnn = True
+
+
+def test_dnn_spatialtf():
+    if not dnn.dnn_available(test_ctx_name):
+        raise SkipTest(dnn.dnn_available.msg)
+
+    utt.seed_rng()
+
+    """
+    Spatial Transformer implementation using Theano from Lasagne
+    Original author: skaae (https://github.com/skaae)
+    """
+    def spatialtf_cpu(theta, inp, scale_height, scale_width, border_mode='nearest'):
```

Comment:
That is correct, the CPU implementation was taken mainly from Lasagne's spatial transformer. I also think it can be ported to Theano, and it would be a good replacement for the third part of GSoC.